osf/external/spam/tasks.py

```diff
@@ -39,11 +39,16 @@ def reclassify_domain_references(notable_domain_id, current_note, previous_note)
 
 
 def _check_resource_for_domains(guid, content):
-    from osf.models import Guid, NotableDomain, DomainReference
-    guid = Guid.load(guid)
-    if not guid:
-        return f'{guid} not found'
-    resource = guid.referent
+    from osf.models import Guid, NotableDomain, DomainReference, VersionedGuidMixin, Preprint
+    if VersionedGuidMixin.GUID_VERSION_DELIMITER in guid:
+        resource = Preprint.load(guid)
```

Comment:
`VersionedGuidMixin.GUID_VERSION_DELIMITER in guid` is not a guarantee that object is `Preprint` because we can't control user input. User can use an existing project guid and appended it with `_v<number>`. However, it doesn't matter here since if user try to do that, we will return `not found` as expected too, which is a similar case to https://openscience.atlassian.net/browse/ENG-6679.

(Just throwing my thoughts out here, your code looks good).